log/format.go

```diff
@@ -262,6 +269,12 @@ func formatLogfmtValue(value interface{}) string {
 		// expensive.
 		return t.Format(timeFormat)
 	}
+	if term {
+		if s, ok := value.(TerminalStringer); ok {
+			// Custom terminal stringer provided, use that
+			return escapeString(s.TerminalString())
+		}
+	}
```

Comment:
Maybe this should be in formatShared? The check for `fmt.Stringer` is also located there.